### PR TITLE
Add support for label instead of comment for OK status

### PR DESCRIPTION
**Pull request by millerdev**

### CODE REVIEW

File: settings.sample.py [hunk 1]
```diff
@@ -73,3 +73,9 @@
 # LINTRC_DEFAULTS = './lintrc_defaults.ini'
 
 OK_COMMENT = ':+1: No lint errors found.'
+
+# Set to True to use a label instead of a comment for OK status.
```
Comment (markstory) [inline settings.sample.py]: Ideally lint-review could create the label if its missing. That would cut down on the work to integrate lint-review with a project.

File: settings.sample.py [hunk 2]
```diff
@@ -73,3 +73,9 @@
...
+# Cuts down on github notification noise.
+# Don't forget to create the label on github!
+NO_OK_NOTIFY = False
```
Comment (markstory) [inline settings.sample.py]: The config options should be in the positive form preferrably. Right now its harder to figure out what `NO_OK = False` will end up doing. If instead the config option was `ADD_OK_LABEL` I think the code and behavior would be easier to figure out.

File: lintreview/review.py [hunk 3]
```diff
@@ -41,6 +41,50 @@ def __repr__(self):
...
+            gh.issues.labels.remove_from_issue(pull_request_number, self.label)
+        except:
+            log.warn("Failed to remove label '%s'", self.label)
+
+    def publish(self, gh, pull_request_number):
+        # remove+add to show latest activity
+        self.remove(gh, pull_request_number)
+        log.debug("Publishing issue label '%s'", self.label)
+        try:
+            #gh.issues.labels.add_to_issue(pull_request_number, [self.label])
```
Comment (markstory) [inline lintreview/review.py]: Why is this commented out?

File: settings.sample.py (hunk 1 above)
Comment (millerdev) [inline settings.sample.py]: Fixed

File: lintreview/review.py (hunk 3 above)
Comment (millerdev) [inline lintreview/review.py]: `add_to_issue` is broken. I added a comment to clarify this. If it's ever fixed this line should replace the following work-around.

File: lintreview/review.py [hunk 4]
```diff
@@ -41,6 +42,59 @@ def __repr__(self):
...
+            if not any(self.label == label.name for label in labels):
+                return
+            log.debug("Removing issue label '%s'", self.label)
+            gh.issues.labels.remove_from_issue(pull_request_number, self.label)
+        except:
+            log.warn("Failed to remove label '%s'", self.label)
+
+    def publish(self, gh, pull_request_number):
+        # remove+add to show latest activity
+        self.remove(gh, pull_request_number)
```
Comment (markstory) [inline lintreview/review.py]: Right now the first time a project is reviewed if the label was not created by hand a warning will be emitted. You could fix that by checking/creating the label as the first step.

Comment (millerdev) [inline lintreview/review.py]: I don't think so, because it lists the labels on the issue first and removes it only if the issue has our label.

Comment (markstory) [inline lintreview/review.py]: Right you are.

**Comment by markstory** [comment]
New tag published :smile:

**Pull request merged**